WMF Framework/Significant Events Endpoint/SignificantEventsModels.swift

```diff
@@ -0,0 +1,616 @@
+                                dict.nonEmptyValueForKey(key: "author-first") ??
+                                dict.nonEmptyValueForKey(key: "first1") ??
+                                dict.nonEmptyValueForKey(key: "given1") ??
+                                dict.nonEmptyValueForKey(key: "author-first1") ??
+                                dict.nonEmptyValueForKey(key: "author1-first")
+                
+                self.yearPublished = dict.nonEmptyValueForKey(key: "year")
+                self.locationPublished = dict.nonEmptyValueForKey(key: "location") ??
+                                            dict.nonEmptyValueForKey(key: "place")
+                
+                self.publisher = dict.nonEmptyValueForKey(key: "publisher") ??
+                                dict.nonEmptyValueForKey(key: "distributor") ??
+                                dict.nonEmptyValueForKey(key: "institution") ??
+                                dict.nonEmptyValueForKey(key: "newsgroup")
+                
+                self.pagesCited = dict.nonEmptyValueForKey(key: "pages") ??
+                    dict.nonEmptyValueForKey(key: "pp")
+                
+                self.isbn = dict.nonEmptyValueForKey(key: "isbn") ??
+                            dict.nonEmptyValueForKey(key: "ISBN13") ??
+                            dict.nonEmptyValueForKey(key: "isbn13") ??
+                            dict.nonEmptyValueForKey(key: "ISBN")
+            }
+        }
+        
+        //https://en.wikipedia.org/wiki/Template:Cite_journal#TemplateData
+        public struct Journal {
+            public let lastName: String?
+            public let firstName: String?
+            public let sourceDateString: String?
+            public let title: String
+            public let journal: String
+            public let urlString: String?
+            public let volumeNumber: String?
+            public let pages: String?
+            public let database: String?
+            
+            init?(dict: [String: String]) {
+                guard let title = dict.nonEmptyValueForKey(key: "title"),
+                let journal = dict.nonEmptyValueForKey(key: "journal") else {
+                    return nil
+                }
+                
+                self.title = title
+                self.journal = journal
+                
+                self.lastName = dict.nonEmptyValueForKey(key: "last") ??
+                dict.nonEmptyValueForKey(key: "author") ??
+                dict.nonEmptyValueForKey(key: "author1") ??
+                dict.nonEmptyValueForKey(key: "authors") ??
+                dict.nonEmptyValueForKey(key: "last1")
+                
+                self.firstName = dict.nonEmptyValueForKey(key: "first") ??
+                dict.nonEmptyValueForKey(key: "first1")
+                
+                self.sourceDateString = dict.nonEmptyValueForKey(key: "date")
+                self.urlString = dict.nonEmptyValueForKey(key: "url")
```

Comment:
I think the doc states this can also be an uppercased `URL`: 

```swift
self.urlString = dict.nonEmptyValueForKey(key: "url") ?? dict.nonEmptyValueForKey(key: "URL") // or something like the `caseInsensitive` approach from above 😉
```

Reply:
👍  Done.